seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -195,6 +217,89 @@ public CoordinatorService(
         masterActiveListener = Executors.newSingleThreadScheduledExecutor();
         masterActiveListener.scheduleAtFixedRate(
                 this::checkNewActiveMaster, 0, 100, TimeUnit.MILLISECONDS);
+        isJobPending =
+                engineConfig.getScheduleStrategy().equals(ScheduleStrategy.WAIT) && !dynamicSlot;
+        if (isJobPending) {
+            logger.info("Start pending job schedule thread");
+            // start pending job schedule thread
+            startPendingJobScheduleThread();
+        }
+    }
+
+    private void startPendingJobScheduleThread() {
+        Runnable pendingJobScheduleTask =
+                () -> {
+                    Thread.currentThread().setName("pending-job-schedule-runner");
+                    while (true) {
+                        if (pendingJob.isEmpty()) {
+                            try {
+                                Thread.sleep(3000);
+                            } catch (InterruptedException e) {
+                                logger.severe(ExceptionUtils.getMessage(e));
+                            }
+                        } else {
+                            pendingJobSchedule();
+                        }
+                    }
+                };
+        executorService.submit(pendingJobScheduleTask);
+    }
+
+    private void pendingJobSchedule() {
+        pendingJob.stream()
+                .findFirst()
+                .ifPresent(
+                        pendingJobEntry -> {
+                            Long jobId = pendingJobEntry._1;
+                            JobMaster jobMaster = pendingJobEntry._2;
+
+                            logger.info(
+                                    "Start calculating whether pending task resources are enough: "
+                                            + jobId);
+
+                            if (jobMaster.preApplyResources()) {
+                                logger.info("Resources enough, start running: " + jobId);
+                                pendingJob.poll();
```

Comment:
We need to add a check to ensure that `pendingJob.stream().findFirst()` and `pendingJob.poll()` return the same jobmaster